### PR TITLE
Add config parameters to support third-party reCAPTCHA v2 compatible APIs

### LINKED ISSUE #16457: Support third-party reCAPTCHA v2 compatible APIs
Support https://www.hcaptcha.com/

Fixed by #16406

### PR DESCRIPTION
**Pull request by jprenken**
# Description

Add config parameters to support third-party reCAPTCHA v2 compatible APIs.

This allows users to switch to hCaptcha (for example) without having to change hard-coded URLs and parameters.

I successfully tested this patch with both Google reCAPTCHA (just by setting `CaptchaLoginPublicKey` & `CaptchaLoginPrivateKey` as normal), and with hCaptcha using these parameters:

```php
$cfg['CaptchaApi'] = 'https://www.hcaptcha.com/1/api.js';
$cfg['CaptchaCsp'] = 'https://hcaptcha.com https://*.hcaptcha.com';
$cfg['CaptchaRequestParam'] = 'h-captcha';
$cfg['CaptchaResponseParam'] = 'h-captcha-response';
$cfg['CaptchaSiteVerifyURL'] = 'https://hcaptcha.com/siteverify';
```

I chose not to expose the new parameters to the features form, since I'm not fully familiar with phpMyAdmin's security model and these settings probably should only be available to administrators.

Fixes: #16457

- [X] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [X] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [X] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [X] Every commit has a descriptive commit message.
- [X] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [X] Any new functionality is covered by tests.

Signed-off-by: James Renken <james@renken.is>

### CODE REVIEW

File: doc/config.rst [hunk 1]
```diff
@@ -1884,6 +1884,43 @@ Cookie authentication options
 
     .. versionadded:: 5.0.3
 
+.. config:option:: $cfg['CaptchaApi']
+
+    :type: string
+    :default: ``'https://www.google.com/recaptcha/api.js'``
+
+    The URL for the reCaptcha v2 service's API, either Google's or a compatible one.
+
```
Comment (williamdes) [inline doc/config.rst:1893]: Could you add some `    .. versionadded:: 5.1.0` lines for each option ?

**Review by williamdes — APPROVED** [review]
This is very good and clean work !

**Comment by williamdes** [comment]
@jprenken It seems you did forget to import you public GPG key into GitHub

**Comment by jprenken** [comment]
Thanks! Apologies, I had signed my first commit with the wrong key. I added a manual Signed-off-by; will that be OK to merge?

**Comment by williamdes** [comment]
> Thanks! Apologies, I had signed my first commit with the wrong key. I added a manual Signed-off-by; will that be OK to merge?

Sure, I will squash them :)
Your last commit has no GPG signature, I do not mind but maybe you would prefer to have it enabled.

**Review by williamdes — APPROVED** [review]
:100:

**Review by MauricioFauth — APPROVED** [review]

**Review by ibennetch — APPROVED** [review]
I agree with the others, this is quite excellent and detail-oriented work. Very nicely done. We'll include this in 5.1.0, which is scheduled for release in less than a month.

**Comment by williamdes** [comment]
Hi @jprenken 

I could not make this work, maybe you could help me ?
```php

$cfg['CaptchaApi'] = 'https://www.hcaptcha.com/1/api.js';
$cfg['CaptchaCsp'] = 'https://hcaptcha.com https://*.hcaptcha.com';
$cfg['CaptchaRequestParam'] = 'h-captcha';
$cfg['CaptchaResponseParam'] = 'h-captcha-response';
$cfg['CaptchaSiteVerifyURL'] = 'https://hcaptcha.com/siteverify';
$cfg['CaptchaLoginPrivateKey'] = '0xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx';
$cfg['CaptchaLoginPublicKey'] = '0xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx';
```

**Comment by williamdes** [comment]
It did work, could you add more documentation and an example because it was not very obvious how to use `h-captcha`.

Especially that `$cfg['CaptchaLoginPrivateKey']` is the secret key and `$cfg['CaptchaLoginPublicKey']` is the site key.

**Pull request merged**